OpenDreamRuntime/Procs/DMOpcodeHandlers.cs

```diff
@@ -1470,7 +1470,35 @@ static class DMOpcodeHandlers {
             if (value.TryGetValueAsString(out string refString)) {
                 if(int.TryParse(refString, out var refID))
                 {
-                    state.Push(new DreamValue(DreamObject.GetFromReferenceID(state.DreamManager, refID)));
+                    // The first digit is the type
+                    var typeId = int.Parse(refString.Substring(0, 1));
+                    refID = int.Parse(refString.Substring(1));
```

Comment:
May be better for code quality to move this to a `LocateRef()` method in DreamManager.